File: WebTurismo/app/Models/Producto.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Producto extends Model
{
    use HasFactory;
    protected $table = 'productos';

    protected $primaryKey = 'id_producto';
    protected $fillable = [
        'id_emprendimiento',
        'id_categoria',
        'nombre_producto',
        'descripcion',
        'precio',
        'stock',
        'imagen_url',
        'estado',
    ];

    //relacion de muchos a uno con categorias
    public function category(){
        return $this->belongsTo(Categoria::class);
    }


}

File: CTProyecto/app/src/main/java/pe/edu/upeu/ctproyecto/data/model/Producto.kt
package pe.edu.upeu.ctproyecto.data.model

import com.google.gson.annotations.SerializedName

data class Producto(
    @SerializedName("id_producto") val id: Int,
    val nombre_producto: String,
    val descripcion: String?,
    val precio: Double,
    val stock: Int,
    val estado: String,
    val imagen_url: String?,
    val id_categoria: Int,
    val id_emprendimiento: Int
)
